non_contextual_word_embeddings.py

```diff
@@ -0,0 +1,115 @@
+from settings import DATA_DIR
+import os
+
+import pandas as pd
+import numpy as np
+from gensim.models import KeyedVectors
+
+def load_gensim_fasttext_model(model_path):
+    """
+    Load the pre-trained fastText model using gensim.
+    Args:
+        model_path (str): Path to the pre-trained .vec file.
+    Returns:
+        model: gensim KeyedVectors model.
+    """
+    print("Loading pre-trained fastText model using gensim...")
+    model = KeyedVectors.load_word2vec_format(model_path, binary=False)
+    return model
+
+def preprocess_text(text):
+    """
+    Additional preprocessing on the text.
+    - Lowercases text
+    - Removes extra spaces
+    Args:
+        text (str): Input text to preprocess.
+    Returns:
+        str: Cleaned text.
+    """
+    text = text.lower()
+    text = ' '.join(text.split())
+    return text
+
+def text_to_embedding(text, model):
+    """
+    Convert a preprocessed text into an averaged fastText embedding using gensim.
+    Args:
+        text (str): Preprocessed input text.
+        model: gensim KeyedVectors model.
+    Returns:
+        np.ndarray: Averaged embedding vector for the text.
+    """
+    words = text.split()
+    embeddings = [model[word] for word in words if word in model]
+    
+    if not embeddings:
+        return np.zeros(model.vector_size)  # Return a zero vector if no words are found in the model
+    return np.mean(embeddings, axis=0)
+
+def process_csv(input_file, output_file, model_path, col1, col2, col3):
+    """
+    Read the CSV, convert text to embeddings, and save to a new CSV.
+    Args:
+        input_file (str): Path to the input CSV file.
+        output_file (str): Path to save the output CSV file.
+        model_path (str): Path to the pre-trained fastText model.
+        col1 (str): Name of the first text column.
```

Comment:
this is not the best way how to pass on multiple arguments of the same type. It is not prepared for scaling or modification. Use `list` instead or `*args` (I don't know exactly how `*args` works, but I think it is suitable for this case when you don't know exactly the number of columns to proceed.).